packages/playground/blueprints/src/lib/steps/install-asset.ts

```diff
@@ -0,0 +1,94 @@
+import type { UniversalPHP } from '@php-wasm/universal';
+import { writeFile } from './client-methods';
+import { unzip } from './import-export';
+
+export interface InstallAssetOptions {
+	/**
+	 * The zip file to install.
+	 */
+	zipFile: File;
+	/**
+	 * Target path to extract the main folder.
+	 * @example
+	 *
+	 * <code>
+	 * const targetPath = `${await playground.documentRoot}/wp-content/plugins`;
+	 * </code>
+	 */
+	targetPath: string;
+}
+
+/**
+ * Install asset: Extract folder from zip file and move it to target
+ */
+export async function installAsset(
+	playground: UniversalPHP,
+	{ targetPath, zipFile }: InstallAssetOptions
+): Promise<{
+	assetFolderPath: string;
+	assetFolderName: string;
+}> {
+	// Extract to temporary folder so we can find asset folder name
+
+	const zipFileName = zipFile.name;
+	const tmpFolder = `/tmp/assets`;
+	const tmpZipPath = `/tmp/${zipFileName}`;
+
+	async function cleanTmpFolder() {
+		await playground.rmdir(tmpFolder, {
+			recursive: true,
+		});
+	}
+
+	if (await playground.fileExists(tmpFolder)) {
+		await cleanTmpFolder();
+	}
+
+	await writeFile(playground, {
+		path: tmpZipPath,
+		data: zipFile,
+	});
+
+	await unzip(playground, {
+		zipPath: tmpZipPath,
+		extractToPath: tmpFolder,
+	});
```

Comment:
didn't we pipe the zip file in another PR straight into `unzip` so we could skip the temporary file? is that something we can do here?

it not only eliminates the need to write a temp file and cleanup (which we're not doing if this crashes during the process) but can also eliminate notable lag, because the unzip process can follow the chunks of data as they are written instead of waiting until they are all finished, and then reading them again.

Reply:
@dmsnell that happens in `wp-now` and cannot be easily done with Blueprints yet because the "resource" abstraction outputs a `File` object. The stream, however, was node-specific. Figuring this out would be a great follow-up, though. Loosely thinking, maybe a "zipped resource type" or the ability to output a stream from a regular resource would do the trick.